Assembly-CSharp/ModLoader.cs

```diff
@@ -23,15 +23,28 @@ namespace Modding
     [PublicAPI]
     internal static class ModLoader
     {
+        [Flags]
+        public enum ModLoadState
+        {
+            NotStarted = 0,
+            Started = 1,
+            Preloaded = 2,
+            Loaded = 4,
+        }
+
+        public static ModLoadState modLoadState = ModLoadState.NotStarted;
+
         /// <summary>
         ///     Checks if the mod loads are done.
         /// </summary>
-        public static bool Loaded;
+        public static bool Loaded => modLoadState.HasFlag(ModLoadState.Loaded);
```

Comment:
The properties are unnecessary given that these are used in all of one spot

Reply:
Actually Preloaded is used in 3 spots. But yeah I can remove them